irohad/multi_sig_transactions/impl/gossip_propagation_strategy.cpp

```diff
@@ -46,26 +46,25 @@ namespace iroha {
   }
 
   bool GossipPropagationStrategy::initQueue() {
-    // Grub peers from the storage
-    auto data_opt = query->getLedgerPeers();
-    if (!data_opt || data_opt->size() == 0) {
-      return false;
-    }
-    last_data = *data_opt;
-
-    // Generate random ordered vector of unique numbers
-    non_visited.resize(last_data.size());
-    std::iota(non_visited.begin(), non_visited.end(), 0);
-    std::random_shuffle(non_visited.begin(), non_visited.end());
-    return true;
+    return static_cast<bool>(
+        query->getLedgerPeers() |
+            [this](PropagationData data) -> nonstd::optional<int> {
+          last_data = data;
+          if (data.size() == 0) return {};
+          this->non_visited.resize(this->last_data.size());
+          std::iota(this->non_visited.begin(), this->non_visited.end(), 0);
+          std::random_shuffle(this->non_visited.begin(),
```

Comment:
`std::random_shuffle` is deprecated in C++14 and removed in C++17. `std::shuffle` can be used instead.
http://en.cppreference.com/w/cpp/algorithm/random_shuffle

Reply:
Nice point, thanks!